interp/src/test.rs

```diff
@@ -1,7 +1,477 @@
+    }
+
+    #[test]
+    fn smoosher_to_hm() {
+        let mut smoosher = Smoosher::new();
+        smoosher.set("alma", Value::try_from_init(18, 32).unwrap());
+        smoosher.new_scope();
+        smoosher.set("jonathan", Value::try_from_init(14, 32).unwrap());
+        smoosher.new_scope();
+        smoosher.set("joseph", Value::try_from_init(19, 32).unwrap());
+        smoosher.new_scope();
+        smoosher.set("joseph", Value::try_from_init(436, 32).unwrap());
+        smoosher.set("ari", Value::try_from_init(12, 32).unwrap());
+        let hm = smoosher.to_hm();
+        //that type annotation seems a bit wack
+        assert_eq!(hm.len(), 4);
+        assert_eq!(hm.get(&"alma").unwrap().as_u64(), 18);
+        assert_eq!(hm.get(&"jonathan").unwrap().as_u64(), 14);
+        assert_eq!(hm.get(&"joseph").unwrap().as_u64(), 436);
+        assert_eq!(hm.get(&"ari").unwrap().as_u64(), 12);
+    }
+
+    #[test]
+    fn smoosher_diff_2() {
+        let mut smoosher = Smoosher::new();
+        smoosher.set("alma", Value::try_from_init(18, 32).unwrap());
+        smoosher.new_scope();
+        smoosher.set("joseph", Value::try_from_init(19, 32).unwrap());
+        smoosher.new_scope();
+        smoosher.set("jonathan", Value::try_from_init(14, 32).unwrap());
+        smoosher.new_scope();
+        smoosher.set("jonathan", Value::try_from_init(15, 32).unwrap());
+        smoosher.new_scope();
+        smoosher.set("alma", Value::try_from_init(19, 32).unwrap());
+        smoosher.set("joseph", Value::try_from_init(19, 32).unwrap());
+        //there are 5 scopes, check diff 2 and see that the resulting hm
+        //has alma, jonathan, but not joseph.
+        let diff_2 = smoosher.diff(2);
+        assert!(diff_2.contains_key(&"alma"));
+        assert!(diff_2.contains_key(&"jonathan"));
+        assert_eq!(diff_2.contains_key(&"joseph"), false);
+    }
+
+    #[test]
+    fn smoosher_diff_other() {
+        let mut smoosher = Smoosher::new();
+        smoosher.set("alma", Value::try_from_init(18, 32).unwrap());
+        smoosher.new_scope();
+        smoosher.set("joseph", Value::try_from_init(19, 32).unwrap());
+        smoosher.new_scope();
+        smoosher.set("jonathan", Value::try_from_init(14, 32).unwrap());
+        let mut smoosher2 = Smoosher::new();
+        smoosher2.set("jonathan", Value::try_from_init(15, 32).unwrap());
+        smoosher2.new_scope();
+        smoosher2.set("alma", Value::try_from_init(19, 32).unwrap());
+        smoosher2.set("joseph", Value::try_from_init(19, 32).unwrap());
+        let diff_2 = smoosher.diff_other(&smoosher2);
+        assert!(diff_2.contains_key(&"alma"));
+        assert_eq!(diff_2.get(&"alma").unwrap().as_u64(), 18);
+        assert!(diff_2.contains_key(&"jonathan"));
+        assert_eq!(diff_2.get(&"jonathan").unwrap().as_u64(), 14);
+        assert_eq!(diff_2.contains_key(&"joseph"), false);
+    }
+}
+
 mod prim_test {
+    #[allow(unused)]
```

Comment:
Are these necessary?

Reply:
yea, otherwise clippy breaks, but if we remove them the tests don't compile. maybe we are putting the imports in the wrong place?

Reply:
Oh, it's because you're using * imports. Better to just import what you need for the tests as a general principal

Reply:
makes sense although for `primitives` doesn't that entail importing `std_reg`, `std_le`, individually, etc?

Reply:
You can use the `import X as Y` syntax to make it a bit cleaner. For example:
```
import primitives::Primitives as P;
P::std_reg(...)
P::std_le(..)
```